builder/dockerfile/parser/parser.go

```diff
@@ -107,29 +108,31 @@ func (d *Directive) setEscapeToken(s string) error {
 	return nil
 }
 
-// processLine looks for a parser directive '# escapeToken=<char>. Parser
-// directives must precede any builder instruction or other comments, and cannot
-// be repeated.
-func (d *Directive) processLine(line string) error {
+// possibleParserDirective looks for one or more parser directives '# escapeToken=<char>' and
+// '# platform=<string>'. Parser directives must precede any builder instruction
+// or other comments, and cannot be repeated.
+func (d *Directive) possibleParserDirective(line string) error {
 	if d.processingComplete {
 		return nil
 	}
-	// Processing is finished after the first call
-	defer func() { d.processingComplete = true }()
 
+	logrus.Debugf("[BUILDER] considering parser directive in %q", line)
 	tecMatch := tokenEscapeCommand.FindStringSubmatch(strings.ToLower(line))
-	if len(tecMatch) == 0 {
-		return nil
-	}
-	if d.escapeSeen == true {
-		return errors.New("only one escape parser directive can be used")
-	}
-	for i, n := range tokenEscapeCommand.SubexpNames() {
-		if n == "escapechar" {
-			d.escapeSeen = true
-			return d.setEscapeToken(tecMatch[i])
+	if len(tecMatch) != 0 {
+		if d.escapeSeen == true {
+			return errors.New("only one escape parser directive can be used")
+		}
+		for i, n := range tokenEscapeCommand.SubexpNames() {
+			if n == "escapechar" {
+				logrus.Debugf("[BUILDER] escape set to '%s'", tecMatch[i])
+				d.escapeSeen = true
+				return d.setEscapeToken(tecMatch[i])
+			}
 		}
 	}
+
+	logrus.Debugf("[BUILDER] no more parser directives allowed from this point")
+	d.processingComplete = true
```

Comment:
Yes, that was deliberate to remove the `defer`.

I'm not sure we necessarily want to match the same regex (actually the one I was contemplating adding was a string, not a single character/rune), but we can cross that bridge one I PR in a second directive. For now, I don't think it makes sense to add a `switch`.